### PR TITLE
Bug 1826781: Added ENV comparison for ES deployment and stateful set

### PR DESCRIPTION
**Pull request by igor-karpukhin**
This ticket fixes these bugs: 
* https://bugzilla.redhat.com/show_bug.cgi?id=1824006
* https://bugzilla.redhat.com/show_bug.cgi?id=1825088

### CODE REVIEW

File: pkg/k8shandler/deployment.go [hunk 1]
```diff
@@ -661,6 +663,12 @@ func (node *deploymentNode) isChanged() bool {
 			changed = true
 		}
 
+		if !comparators.EnvValueEqual(desiredContainer.Env, nodeContainer.Env) {
```
Comment (syedriko) [inline pkg/k8shandler/deployment.go:666]: There's a comment a few lines prior saying "// Only Image and Resources (CPU & memory) differences trigger rolling restart". Does it change with the addition of the env var comparison?

File: pkg/k8shandler/statefulset.go [hunk 2]
```diff
@@ -641,6 +644,12 @@ func (node *statefulSetNode) isChanged() bool {
 			changed = true
 		}
 
+		if !comparators.EnvValueEqual(desiredContainer.Env, nodeContainer.Env) {
```
Comment (syedriko) [inline pkg/k8shandler/statefulset.go:647]: Same observation as in deployment.go about the "// Only Image and Resources..." comment

File: pkg/k8shandler/deployment.go (hunk 1 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/deployment.go:666]: Yes, that will also trigger a rolling restart.

File: pkg/k8shandler/statefulset.go (hunk 2 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/statefulset.go:647]: Yes, that will also trigger a rolling restart.

File: pkg/k8shandler/deployment.go (hunk 1 above)
Comment (syedriko) [reply pkg/k8shandler/deployment.go:666]: Should the comment reflect that? :)

**Pull request merged**